apps/desktop-wallet/src/contexts/global.tsx

```diff
@@ -82,58 +82,73 @@ export const GlobalContextProvider: FC<{ overrideContextValue?: PartialDeep<Glob
   const triggerNewVersionDownload = () => setNewVersionDownloadTriggered(true)
   const resetNewVersionDownloadTrigger = () => setNewVersionDownloadTriggered(false)
 
-  const unlockWallet = async ({ event, walletId, password, afterUnlock, passphrase }: WalletUnlockProps) => {
-    let wallet
+  const unlockWallet = async (props: WalletUnlockProps | null) => {
+    if (!props) return
+
+    const { event, walletId, afterUnlock } = props
+    let { password, passphrase } = props
+    let encryptedWallet: StoredEncryptedWallet | null
+    let initialAddress: NonSensitiveAddressData
+    let version: EncryptedMnemonicVersion
 
     try {
-      wallet = WalletStorage.load(walletId, password)
+      encryptedWallet = walletStorage.load(walletId)
+      version = keyring.initFromEncryptedMnemonic(encryptedWallet.encrypted, password, passphrase ?? '')
```

Comment:
I agree that it's not clear what the structure of the JSON is. I also want to improve this. Not sure `encryptedMnemonic` is the way to go. I'll do some research.

Reply:
I added a comment, let me know if it helps: 95cdbf6bf (in PR #504)

I'd prefer to avoid changing the structure of the stored wallet once again, to not have to write yet another migration.

Also, if in the future we want to store additional encrypted data alongside the mnemonic, then the name would be misleading and we'd need a migration again.

Reply:
fine by me, thanks for adding the comment.